pkgs/development/python-modules/pymonocypher/default.nix

```
@@ -0,0 +1,44 @@
+{
+  buildPythonPackage,
+  cython,
+  fetchFromGitHub,
+  lib,
+  monocypher,
+  setuptools,
+}:
+let
+  version = "4.0.2.5";
+in
+buildPythonPackage {
+  pname = "pymonocypher";
+  inherit version;
+
+  pyproject = true;
+  build-system = [
+    cython
+    setuptools
+  ];
+
+  src = fetchFromGitHub {
+    owner = "jetperch";
+    repo = "pymonocypher";
+    rev = "v${version}";
+    hash = "sha256-3vnF2MnrjI7pRiOTjPZ0D8tDojfdGJ2kSlLqF7Kkp5Y=";
+  };
+
+  buildInputs = [ monocypher ];
+
+  doCheck = true;
+
+  pythonImportsCheck = [ "monocypher" ];
```

You need to add `unittestCheckHook` for tests on `checkPhase`:
```suggestion
  doCheck = true;

  pythonImportsCheck = [ "monocypher" ];
  
  nativeCheckInputs = [
    unittestCheckHook
  ];
```

---

Thank you, @renesat! Good now? Build log shows

```
python3.12-pymonocypher> Executing unittestCheckPhase
python3.12-pymonocypher> unittest flags: ''
python3.12-pymonocypher> ..../nix/store/fqm9bqqlmaqqr02qbalm1bazp810qfiw-python3-3.12.9/lib/python3.12/unittest/case.py:589: DeprecationWarning: Provide the full 64-byte key from generate_signing_key_pair()
python3.12-pymonocypher>   if method() is not None:
python3.12-pymonocypher> ..........
python3.12-pymonocypher> ----------------------------------------------------------------------
python3.12-pymonocypher> Ran 14 tests in 0.096s
python3.12-pymonocypher>
python3.12-pymonocypher> OK
python3.12-pymonocypher> Finished executing unittestCheckPhase
```